server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -59,6 +59,14 @@ public class QueryPhase {
     public QueryPhase() {}
 
     public static void execute(SearchContext searchContext) throws QueryPhaseExecutionException {
+        if (searchContext.rankShardContext() != null) {
+            searchContext.rankShardContext().executeQueries(searchContext);
+        } else {
+            executeInternal(searchContext);
+        }
+    }
+
+    public static void executeInternal(SearchContext searchContext) throws QueryPhaseExecutionException {
```

Comment:
I renamed this to executeQuery. It's still a bit more complicated than previously, but all the logic is now contained in QueryPhase instead of externally which seems like a big improvement.